hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILECrck.py

```diff
@@ -0,0 +1,156 @@
+
+from health_ml.networks.layers.attention_layers import GatedAttentionLayer
+from health_ml.utils import fixed_paths
+
+from histopathology.configs.classification.BaseMIL import BaseMIL
+from histopathology.datamodules.base_module import TilesDataModule
+from histopathology.datamodules.tcga_crck_module import TcgaCrckTilesDataModule
+from histopathology.models.encoders import (
+    HistoSSLEncoder,
+    ImageNetEncoder,
+    ImageNetSimCLREncoder,
+    SSLEncoder,
+)
+from histopathology.models.transforms import EncodeTilesBatchd, LoadTilesBatchd
+from histopathology.datasets.tcga_crck_tiles_dataset import TcgaCrck_TilesDataset
+
+
+class DeepSMILECrck(BaseMIL):
+    def __init__(self, **kwargs: Any) -> None:
+        # Define dictionary with default params that can be overridden from subclasses or CLI
+        default_kwargs = dict(
+            # declared in BaseMIL:
+            pooling_type=GatedAttentionLayer.__name__,
+            # declared in DatasetParams:
+            local_datasets=[Path("/tmp/datasets/TCGA-CRCk")],
+            azure_datasets=["TCGA-CRCk"],
+            # To mount the dataset instead of downloading in AML, pass --use_dataset_mount in the CLI
+
+            # declared in TrainerParams:
+            max_epochs=16,
+
+            # declared in WorkflowParams:
+            # number_of_cross_validation_splits=5,
+            # cross_validation_split_index=0,
+            # declared in OptimizerParams:
+            l_rate=5e-4,
+            weight_decay=1e-4,
+            adam_betas=(0.9, 0.99),
+        )
+        default_kwargs.update(kwargs)
+        super().__init__(**default_kwargs)
+
+        self.best_checkpoint_filename = "checkpoint_max_val_auroc"
+        self.best_checkpoint_filename_with_suffix = (
+            self.best_checkpoint_filename + ".ckpt"
+        )
+        self.checkpoint_folder_path = "outputs/checkpoints/"
+
+        best_checkpoint_callback = ModelCheckpoint(
+            dirpath=self.checkpoint_folder_path,
+            monitor="val/auroc",
+            filename=self.best_checkpoint_filename,
+            auto_insert_metric_name=False,
+            mode="max",
+        )
+        self.callbacks = best_checkpoint_callback
+
+    @property
+    def cache_dir(self) -> Path:
+        return Path(
+            f"/tmp/innereye_cache1/{self.__class__.__name__}-{self.encoder_type}/"
+        )
+
+    def setup(self) -> None:
+        if self.encoder_type == SSLEncoder.__name__:
+            from InnerEye.ML.configs.histo_configs.run_ids import innereye_ssl_checkpoint_crck_4ws
+            self.downloader = CheckpointDownloader(
+                azure_config_json_path=get_workspace(),
+                run_id=innereye_ssl_checkpoint_crck_4ws,
+                checkpoint_filename="best_checkpoint.ckpt",
+                download_dir="outputs/",
+                remote_checkpoint_dir=Path("outputs/checkpoints")
+            )
+            os.chdir(fixed_paths.repository_parent_directory())
```

Comment:
what is the relative position of the path with respect to this config? if different from what we had in the configuration before this will not work

Reply:
Of the checkpoint dir? It is in root_dir -> outputs -> checkpoints, same as before